server/src/main/java/org/elasticsearch/search/rank/context/RankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -44,6 +45,16 @@ public RankFeaturePhaseRankCoordinatorContext(int size, int from, int rankWindow
      */
     protected abstract void computeScores(RankFeatureDoc[] featureDocs, ActionListener<float[]> scoreListener);
 
+    /**
+     * Filters and sorts the provided documents.
+     * @param originalDocs documents to process
+     */
+    protected RankFeatureDoc[] filterAndSort(RankFeatureDoc[] originalDocs) {
```

Comment:
Minor, but maybe we could rename this as in this base class there's no filtering taking place.

Reply:
Renamed to `preprocess()` and added Javadoc (https://github.com/elastic/elasticsearch/pull/109813/commits/3b18dc0fe7f19d1614f7a6ad14cfe6b1948500f1).